azext_edge/edge/providers/support/base.py

```diff
@@ -218,81 +188,34 @@ def process_services(
     field_selector: str = None,
     prefix_names: List[str] = None,
 ):
```

Comment:
nit, we can add type hints for return types since we spending time improving base